pages/api/gas.ts

```diff
@@ -0,0 +1,35 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+export default async (req: NextApiRequest, res: NextApiResponse) => {
+  if (req.method !== "GET") {
+    return res.status(400).json({ error: "invalid method" });
+  }
+
+  const gasPriceEndpoint = `https://api.etherscan.io/api?module=gastracker&action=gasoracle&apikey=${process.env.ETHERSCAN_KEY}`;
+  const ethPriceEndpoint = `https://api.etherscan.io/api?module=stats&action=ethprice&apikey=${process.env.ETHERSCAN_KEY}`;
+
+  const gasPrice = fetch(gasPriceEndpoint);
+  const ethPrice = fetch(ethPriceEndpoint);
+
+  try {
+    const [gasPriceRes, ethPriceRes] = await Promise.all([gasPrice, ethPrice]);
+    const gasData = await gasPriceRes.json();
+    const ethPriceData = await ethPriceRes.json();
+
+    if (gasData.status !== "1") {
+      return res.status(500).json({ error: "Failed to fetch gas price" });
```

Comment:
this should not be a status code , if we get here it means the responses didn't fail so there isn't a server error it's just a data error in some way, so I'd say maybe 400

Reply:
I just changed this, tell me if this is what you had in mind